src/Security/Authentication/Core/src/RemoteAuthenticationHandler.cs

```diff
@@ -285,5 +297,25 @@ protected virtual async Task<HandleRequestResult> HandleAccessDeniedErrorAsync(A
 
             return HandleRequestResult.NoResult();
         }
+
+        protected virtual Endpoint CreateEndpoint<THandler>(PathString path, string name, string httpMethod, Func<THandler, Task> func)
```

Comment:
Looks like the logic for CreateEndpoint is not actually specific to the Security/auth stack, that means we can likely promote this to something at the Http/Auth layer

Reply:
Agreed, this should not be on the handler, handlers are request artifacts. Endpoints are startup configuration, you need to be able to create the endpoint from the scheme/options alone.